ports/servoshell/parser.rs

```diff
@@ -66,10 +66,14 @@ pub(crate) fn location_bar_input_to_url(request: &str, searchpage: &str) -> Opti
     ServoUrl::parse(request).ok().or_else(|| {
         if request.starts_with('/') {
             ServoUrl::parse(&format!("file://{}", request)).ok()
-        } else if request.contains('/') || is_reg_domain(request) {
+        } else if request.contains('/') || is_reg_domain(request) || is_domain_like(request) {
             ServoUrl::parse(&format!("https://{}", request)).ok()
```

Comment:
this seems to regress location bar inputs like `3.5 kg in lb`, which i would expect to go to a search page but now does nothing. i think we take this branch, fail to parse it as a url, return None, then the caller warns “failed to parse location”. some ideas that might help:

- i don’t think we should ever return None for location bar inputs, unless the input was completely empty
- all major browsers seem to do a search when the location bar input contains spaces but is not a complete url (but they don’t have a consistent behaviour for this situation on the command line)

Reply:
For regression, I added `!s.contains(' ')` rule in `is_domain_like`.

Also add `!request.contains(' ')` before evaluating `is_reg_domain(request)` as
`is_reg_domain("3.5 kg in lb")` evaluates true.

> i don’t think we should ever return None for location bar inputs

Agree. I can work on this if everything else looks good.